dataprofiler/data_readers/data_utils.py

```diff
@@ -303,6 +302,43 @@ def detect_file_encoding(file_path, buffer_size=1024, max_lines=20):
     # Typical file representation is utf-8 instead of ascii, treat as such.
     if not encoding or encoding == 'ascii':
         encoding = 'utf-8'
+
+    # Check if encoding can be used to decode without throwing an error
+    def _decode_is_valid(encoding):
+        try: 
+            open(file_path, "rb").read().decode(encoding.lower())
+            return True
+        except: return False
+
+    if not _decode_is_valid(encoding):
+        try:
+            from charset_normalizer import CharsetNormalizerMatches as CnM
+            
+            # Try with small sample 
+            with open(file_path, 'rb') as input_file:
+                raw_data = input_file.read(2560)
+                result = CnM.from_bytes(raw_data, steps=5, 
+                                        chunk_size=512, threshold=0.2,
+                                        cp_isolation=None, cp_exclusion=None,
+                                        preemptive_behaviour=True, explain=False)
+            encoding = result.best().first().encoding
+
+            # Try again with full sample
+            if not _decode_is_valid(encoding): 
+                with open(file_path, 'rb') as input_file:
+                    raw_data = input_file.read(max_lines*buffer_size)
+                    result = CnM.from_bytes(raw_data, steps=max_lines, 
+                                            chunk_size=buffer_size, threshold=0.2,
+                                            cp_isolation=None, cp_exclusion=None,
+                                            preemptive_behaviour=True, explain=False)
+                encoding = result.best().first().encoding
```

Comment:
@lettergram should this be a while loop that doubles the byte size each time if fails until feof or success?